jetty-security/src/main/java/org/eclipse/jetty/security/SecurityHandler.java

```diff
@@ -572,6 +572,11 @@ else if (authentication instanceof Authentication.Deferred)
                             authenticator.secureResponse(request, response, isAuthMandatory, null);
                     }
                 }
+                else if ((authentication == Authentication.UNAUTHENTICATED) && isAuthMandatory)
```

Comment:
Isn't this just:
```suggestion
                else if (isAuthMandatory)
```

ie. if auth is mandatory, then we really need either `ResponseSent` or `User` auth types.